### PR TITLE
Mz/prf sha3

### PR DESCRIPTION
**Pull request by mayeul-zama**

### CODE REVIEW

File: tfhe/src/shortint/oprf.rs [hunk 1]
```diff
@@ -85,7 +108,11 @@ impl ServerKey {
             "The number of random bits asked for (={random_bits_count}) is bigger than full_bits_count (={full_bits_count})"
         );
 
-        let ct = self.create_random_from_seed(seed);
+        assert_eq!(self.pbs_order, PBSOrder::KeyswitchBootstrap);
```
Comment (IceTDrinker) [inline tfhe/src/shortint/oprf.rs:111]: why the assert ? We could keyswitch to get back to the right key size for the PBS_KS case right ?

File: tfhe/src/shortint/oprf.rs [hunk 2]
```diff
@@ -205,7 +252,7 @@ pub(crate) mod test {
     fn oprf_test_uniformity_ci_run_filter() {
         let sample_count: usize = 100_000;
 
-        let p_value_limit: f64 = 0.001;
+        let p_value_limit: f64 = 0.000_001;
```
Comment (IceTDrinker) [inline tfhe/src/shortint/oprf.rs:255]: removing the keyswitch makes it work better ?

**Review by IceTDrinker — COMMENTED** [review]
Looks generally good a few small question

File: tfhe/src/shortint/oprf.rs (hunk 2 above)
Comment (mayeul-zama) [reply tfhe/src/shortint/oprf.rs:255]: No, it's unrelated

File: tfhe/src/shortint/oprf.rs (hunk 1 above)
Comment (mayeul-zama) [reply tfhe/src/shortint/oprf.rs:111]: Yes, looks like it would be ok

File: tfhe/src/shortint/oprf.rs (hunk 2 above)
Comment (IceTDrinker) [reply tfhe/src/shortint/oprf.rs:255]: ok, do you think the limit is better as it was or as it is now ?

Comment (mayeul-zama) [reply tfhe/src/shortint/oprf.rs:255]: Better now as this test should fail with `p_value_limit` probability

Comment (IceTDrinker) [reply tfhe/src/shortint/oprf.rs:255]: ok did you stress test it to check it did not fail in the CI ?

Comment (mayeul-zama) [reply tfhe/src/shortint/oprf.rs:274]: This new bound is less strict (by a huge ratio) than the old one which IIRC only failed one on the CI (p=1/1000) so this should be fine

Comment (IceTDrinker) [reply tfhe/src/shortint/oprf.rs:274]: got it, is it still significant enough ?

File: tfhe/src/shortint/oprf.rs [hunk 3]
```diff
@@ -96,20 +122,54 @@ impl ServerKey {
...
+        lwe_ciphertext_plaintext_add_assign(&mut ct, Plaintext((p - 1) * delta / 2));
+
+        let ct = match self.pbs_order {
+            PBSOrder::KeyswitchBootstrap => ct,
+            PBSOrder::BootstrapKeyswitch => {
+                let mut ct_ksed = allocate_and_trivially_encrypt_new_lwe_ciphertext(
+                    in_lwe_size,
+                    Plaintext(0),
+                    self.ciphertext_modulus,
+                );
```
Comment (IceTDrinker) [inline tfhe/src/shortint/oprf.rs:148]: let mut ct_ksed = LweCiphertext::new(0u64, in_lwe_size, self.ciphertext_modulus);

File: tfhe/src/shortint/oprf.rs [hunk 4]
```diff
@@ -96,20 +122,54 @@ impl ServerKey {
         let acc: LookupTableOwned =
             self.generate_lookup_table_no_encode(|x| (2 * (x / poly_delta) + 1) * delta / 2);
 
-        let mut ct = self.apply_lookup_table(&ct, &acc);
+        let out_lwe_size = self.bootstrapping_key.output_lwe_dimension().to_lwe_size();
 
-        lwe_ciphertext_plaintext_add_assign(&mut ct.ct, Plaintext((p - 1) * delta / 2));
+        let mut ct = allocate_and_trivially_encrypt_new_lwe_ciphertext(
```
Comment (IceTDrinker) [inline tfhe/src/shortint/oprf.rs:127]: same here you can allocate directly a new 0ed ciphertext

File: tfhe/src/shortint/oprf.rs (hunk 2 above)
Comment (mayeul-zama) [reply tfhe/src/shortint/oprf.rs]: Moved this change to https://github.com/zama-ai/tfhe-rs/pull/1179

File: tfhe/src/shortint/oprf.rs (hunk 4 above)
Comment (mayeul-zama) [reply tfhe/src/shortint/oprf.rs]: Done

File: tfhe/src/shortint/oprf.rs (hunk 3 above)
Comment (mayeul-zama) [reply tfhe/src/shortint/oprf.rs]: Done

**Review by IceTDrinker — APPROVED** [review]
thx

**Pull request merged**